MagicPowerAlert.sh

```diff
@@ -5,29 +5,87 @@
 # 2. Add a cron entry to run it, for example:
 #   30 9 * * * /Users/dougb/MagicPower/MagicPowerAlert.sh
 #
+# 3. Or, run frequently but set the muting and inactivity vars
+#   */30 * * * * /Users/dougb/MagicPower/MagicPowerAlert.sh
+#
+#   MUTE_CONSECUTIVE_ALERTS_HOURS=4
+#   INACTIVITY_THRESHOLD_MINS=5
 
-# You can monitor the battery levels and charging status by passing 'status' as
-# the first argument to the script
-getStatus="false"
-
-# You can change the default threshold by passing a value as the first argument to the script
+# You can change the default (20) threshold by passing a value as the first argument to the script
 THRESHOLD="${1:-20}"
 
-# Strip a percent sign in case the user supplied one
-THRESHOLD="${THRESHOLD/\%/}"
+# Set to mute consecutive alerts for a period of minutes. Makes it practical to schedule
+# MagicPowerAlert to run frequently, say every 30 minutes. Really, two alerts per day are
+# probably enough of a reminder. So a good value is maybe 4 hours.
+MUTE_CONSECUTIVE_ALERTS_HOURS=4
+
+# Inactivity detection. If user is afk, don't continue and possibly spam the desktop with alerts.
+# For example, no alerts during the night when running other automated tasks.
+INACTIVITY_THRESHOLD_MINS=5
+
+# You can change the message, if coffee is not your thing
+MESSAGE="Get a coffee and charge:\n"
+
+# set to 1, creates a log file for debugging. Log file is ./MagicPowerAlert.sh.log
+LOGFILE=0
 
+# Change nothing below here
+messages=()
+
+# Crude but simple logger to file.
+function logger() {
+    if [[ $LOGFILE -eq 1 ]]; then
+        echo "$(date) $1" >> ${BASH_SOURCE[0]}.log
```

Comment:
I'd double quote the log file name just in case to avoid any possible word-splitting